pennylane_lightning/src/Gates.cpp

```diff
@@ -556,6 +556,52 @@ void Pennylane::CSWAPGate::applyKernel(const StateVector& state, const std::vect
 
 // -------------------------------------------------------------------------------------------------------------
 
+const string Pennylane::Unitary::label = "QubitUnitary";
+
+Pennylane::Unitary Pennylane::Unitary::create(const vector<double>& parameters) {
+    auto size = parameters.size();
+    unsigned int log2size = 0;
+    while (size >>= 1) ++log2size;
+    unsigned int numQubits = (log2size - 1) / 2;
+
+    return Pennylane::Unitary(numQubits, parameters);
+}
+
+Pennylane::Unitary::Unitary(const int numQubits, const std::vector<double>& real_matrix)
+    : AbstractGate(numQubits)
+    , real_matrix{real_matrix}
+{}
+
+void Pennylane::Unitary::applyKernel(const StateVector& state, const std::vector<size_t>& indices, const std::vector<size_t>& externalIndices) {
+    const vector<double>& real_matrix = asRealMatrix();
```

Comment:
Depending on the relative order of the PRs, inverse computations would be required here as per #89?